### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by kanagnostopoulos**
Refactoring of the author name disambiguation project. Our files lie within src folder except src/s2and which is what S2 had already implemented.

### CODE REVIEW

File: src/s2and_ext/clustering_models.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        pred_clusters = {}
+        uknown_id = 0
+        for signatures in block_dict.values():
+            for signature in signatures:
+                if signature not in self.sign_to_pred_cluster:
+                    pred_clusters[f'uknown_{uknown_id}'] = [signature]
+                    uknown_id += 1
+                else:
+                    cluster = self.sign_to_pred_cluster[signature]
+                    if cluster not in pred_clusters:
```
Comment (CoGian) [inline src/s2and_ext/clustering_models.py:163]: We could use defaultdict to avoid this if..else.

**Review by CoGian — APPROVED** [review]
Great job @kanagnostopoulos !! One question: Have we run hyperparameter tuning for the clusterers ?

File: src/s2and_ext/clustering_models.py (hunk 1 above)
Comment (kanagnostopoulos) [reply src/s2and_ext/clustering_models.py:163]: True, I will make the change, thanks!

Comment (kanagnostopoulos) [reply src/s2and_ext/clustering_models.py:163]: Regarding param tuning of clusters, maybe you mean this: https://github.com/atypon/S2AND/blob/00d7c5c34b7baf6d888ec31573ef83c018c9ce93/src/run_clustering.py#L23 ?

Comment (CoGian) [reply src/s2and_ext/clustering_models.py:163]: Oh yeah, sorry many changed files I was lost :P

Comment (kanagnostopoulos) [reply src/s2and_ext/clustering_models.py:163]: Check out the __call__ of the objective. We were discussing this for optuna optimization without the need of declaring the function in the same script using global variables.

**Review by mbizimis — APPROVED** [review]

**Review by CoGian — COMMENTED** [review]
Everything looks great, merging is fine by me.

**Pull request merged**